sqlglot/generator.py

```diff
@@ -2367,6 +2411,15 @@ def matchagainst_sql(self, expression: exp.MatchAgainst) -> str:
     def jsonkeyvalue_sql(self, expression: exp.JSONKeyValue) -> str:
         return f"{self.sql(expression, 'this')}{self.JSON_KEY_VALUE_PAIR_SEP} {self.sql(expression, 'expression')}"
 
+    def jsonpath_sql(self, expression: exp.JSONPath) -> str:
+        path = generate_json_path(
+            expression.expressions,
+            mapping=self._JSON_PATH_MAPPING,
+            unsupported_callback=self.unsupported,
+        )
+        path = path[1:] if path.startswith(".") else path
```

Comment:
when would path start with .? you could just do lstrip(".")

Reply:
For example when you parse `SELECT JSON('{"fruit":"banana"}') -> 'fruit'` in duckdb and you wanna generate that in snowflake, you'd get a prefix `.` without a dollar sign (since the root is not generated)